.github/workflows/main.yml

```diff
@@ -1,53 +1,80 @@
+# actions can be run locally using act and docker, on Fedora 37 also with podman, using:
+# https://github.com/nektos/act
+# sudo dnf install -y act-cli podman-docker
+# act --bind --container-daemon-socket $XDG_RUNTIME_DIR/podman/podman.sock -W .github/workflows/main.yml
+
 name: Unit tests
 
+# Checks can be skipped by adding "skip-checks: true" to a commit message,
+# or requested by adding "request-checks: true" if disabled by default for pushes:
+# https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/about-status-checks#skipping-and-requesting-checks-for-individual-commits
 on: [push, pull_request]
+env:
+  PYTHONWARNINGS: "ignore"
+  PIP_ROOT_USER_ACTION: "ignore"   # For local testing using act-cli
+  PIP_NO_WARN_SCRIPT_LOCATION: "0" # For local testing using act-cli
 
 jobs:
   test:
     strategy:
+      # max-parallel: 1  # See: https://github.com/xenserver/python-libs/pull/26#discussion_r1179482169
       matrix:
         include:
+        - python-version: '3.11'
+          os: ubuntu-latest
+        - python-version: '3.10'
+          os: ubuntu-latest
         - python-version: '2.7'
           os: ubuntu-20.04
-        - python-version: '3'
+        - python-version: '3.9'
+          os: ubuntu-latest
+        - python-version: '3.8'
           os: ubuntu-latest
+        - python-version: '3.7'
+          os: ubuntu-latest
+        - python-version: '3.6'
+          os: ubuntu-20.04
     runs-on: ${{ matrix.os }}
 
     steps:
       - uses: actions/checkout@v3
         with:
-          fetch-depth: 0
+          fetch-depth: 0  # Needed by diff-cover to get the changed lines: origin/master..HEAD
```

Comment:
diff-quality isn't used anymore, this can be dropped as it makes checkouts more costly